### PR TITLE
Add buildsystem using CMake

### PR DESCRIPTION
**Pull request by MarcinPawlowski1987**
Home work done. (TODO: testing)

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: debuggers
🏅 Granted 10 XP for @MarcinPawlowski1987

**Comment by MarcinPawlowski1987** [comment]
@ziobron , nie bardzo rozumiem o co tutaj chodzi:
🔍 CHECKING: source files list in the variable named with UPPERCASE_WITH_UNDERSCORE convention
oraz
❌ FAILED ON 1 binary with tests exist

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+set(PROJECT_NAME_TEST ${PROJECT_NAME}_GT)
+
+# set(TEST_SRC_LIST test/PizzeriaTest.cpp)
+
+# add tests sources as library
+#add_library(${PROJECT_NAME_TEST}_LIB STATIC ${TEST_SRC_LIST})
+#target_compile_options(${PROJECT_NAME_TEST}_LIB PRIVATE ${FLAGS})
+#target_compile_features(${PROJECT_NAME_TEST}_LIB PRIVATE cxx_std_17)
+
+add_executable(${PROJECT_NAME_TEST} test/PizzeriaTest.cpp)
```
Comment (ziobron) [inline CMakeLists.txt:50]: Your solution is ok, but our checker requires test bin name to be `${PROJECT_NAME}-ut`. I've updated it to work for your solution as well.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+FetchContent_MakeAvailable(googletest)
+
+
+find_library(PTHREAD_LIB pthread)
+
+set(SRC_LIST
+    src/Funghi.cpp
+    src/Margherita.cpp
+    src/Pizza.cpp
+    src/Pizzeria.cpp)
```
Comment (ziobron) [inline CMakeLists.txt:21]: It looks that we have a problem with parser. I have updated the rules for it and now it passes. Your solution is correct.

**Comment by ziobron** [comment]
✅ CI Passed - Task: checker
🏅 Granted 10 XP for @MarcinPawlowski1987

**Review by ziobron — APPROVED** [review]

**Pull request closed without merging**